compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -420,34 +420,35 @@ private void defineConstructs(BLangPackage pkgNode, SymbolEnv pkgEnv) {
         // Treat constants and type definitions in the same manner, since constants can be used as
         // types. Also, there can be references between constant and type definitions in both ways.
         // Thus visit them according to the precedence.
-        List<BLangNode> typeAndClassDefs = new ArrayList<>();
-        pkgNode.constants.forEach(constant -> typeAndClassDefs.add(constant));
-        pkgNode.typeDefinitions.forEach(typDef -> typeAndClassDefs.add(typDef));
+        List<BLangNode> moduleDefs = new ArrayList<>();
+        pkgNode.constants.forEach(constant -> moduleDefs.add(constant));
+        pkgNode.typeDefinitions.forEach(typDef -> moduleDefs.add(typDef));
+        pkgNode.xmlnsList.forEach(xmlns -> moduleDefs.add(xmlns));
```

Comment:
This doesn't handle the second comment in the [issue](https://github.com/ballerina-platform/ballerina-lang/issues/35452), does it?

> When fixing this, make sure the following line in [module-xmlns-decl](https://ballerina.io/spec/lang/master/#module-xmlns-decl) does work. If there's no test to test this add one also.
>
> > It applies only to the source part in which it occurs, as with an import-decl.

proj

- a.bal

```ballerina
import ballerina/io;

public function f1() {
    string exdoc = ex:doc;
    io:println(exdoc);
}

xmlns "http://example.com" as ex;
```

- b.bal

```ballerina
import ballerina/io;

public function f2() {
    string exdoc = ex:doc;
    io:println(exdoc);
}
```

There's no error for `ex:doc`.

Similarly, when b.bal is instead

```ballerina
xmlns "http://example.com" as ex;
```

there's a redeclared symbol error. This seems to be the case on U8 too, and is incorrect.